src/app/utils/field-validation/validators/checkboxValidator.ts

```diff
@@ -0,0 +1,168 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import { ProcessedCheckboxResponse } from 'src/app/modules/submission/submission.types'
+import { ICheckboxField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+
+import { isOtherOption } from './options'
+
+type CheckboxValidator = ResponseValidator<ProcessedCheckboxResponse>
+type CheckboxValidatorConstructor = (
+  checkboxField: ICheckboxField,
+) => CheckboxValidator
+
+/**
+ * Returns a validator to check if answerArray is empty
+ */
+const checkboxAnswerValidator: CheckboxValidator = (response) => {
+  const { answerArray } = response
+
+  return answerArray.length === 0
+    ? left(`CheckboxValidator:\t Answer is empty array`)
+    : right(response)
+}
+
+/**
+ * Returns a validation function to check if number of
+ * selected checkbox options is less than the minimum number specified.
+ */
+const minOptionsValidator: CheckboxValidatorConstructor = (checkboxField) => (
+  response,
+) => {
+  const { validateByValue } = checkboxField
+  const { customMin } = checkboxField.ValidationOptions
+  const { answerArray } = response
+
+  if (!validateByValue || !customMin) return right(response)
+
+  return answerArray.length >= customMin
+    ? right(response)
+    : left(
+        `CheckboxValidator:\t answer has less options selected than minimum specified`,
+      )
+}
+
+/**
+ * Returns a validation function to check if number of
+ * selected checkbox options is more than the maximum number specified.
+ */
+const maxOptionsValidator: CheckboxValidatorConstructor = (checkboxField) => (
+  response,
+) => {
+  const { validateByValue } = checkboxField
+  const { customMax } = checkboxField.ValidationOptions
+  const { answerArray } = response
+
+  if (!validateByValue || !customMax) return right(response)
+
+  return answerArray.length <= customMax
+    ? right(response)
+    : left(
+        `CheckboxValidator:\t answer has more options selected than maximum specified`,
+      )
+}
+
+/**
+ * Returns a validation function to check if the
+ * answers are all either within the specified field options or
+ * have the correct format for 'others' answer, if others is enabled
+ */
+const validOptionsValidator: CheckboxValidatorConstructor = (checkboxField) => (
+  response,
+) => {
+  const { fieldOptions, othersRadioButton } = checkboxField
+  const { answerArray } = response
+
+  return answerArray.every(
+    (answer) =>
+      fieldOptions.includes(answer) || isOtherOption(othersRadioButton, answer),
+  )
+    ? right(response)
+    : left(`CheckboxValidator:\t answer is not valid`)
+}
+
```

Comment:
could you add some explanation here about the thinking behind splitting into "others" vs "non-others" options? I just spent a long time convincing myself that this code covers all the cases correctly, and I'm hoping future devs won't have to do the same.

this is how I'm thinking about it:
```
We split the answers into:
1. Those beginning with "Others: "
2. Those not beginning with "Others: ".

For group 1, we allow at most 1 answer which is not one of the `fieldOptions` if `othersRadioButton` is true, and 0 otherwise. Amongst the answers in the `fieldOptions`, there can be no duplicates.

For group 2, there can be no duplicates.
```